test/components/structures/TimelinePanel-test.js

```diff
@@ -207,24 +208,27 @@ describe('TimelinePanel', function() {
     });
 
     it("should let you scroll down again after you've scrolled up", function(done) {
-        var N_EVENTS = 600;
+        var TIMELINE_CAP = 100; // needs to be more than we can fit in the div
+        var N_EVENTS = 120;     // needs to be more than TIMELINE_CAP
 
         // sadly, loading all those events takes a while
-        this.timeout(N_EVENTS * 30);
+        this.timeout(N_EVENTS * 50);
 
         // client.getRoom is called a /lot/ in this test, so replace
         // sinon's spy with a fast noop.
         client.getRoom = function(id) { return null; };
 
         // fill the timeline with lots of events
         for (var i = 0; i < N_EVENTS; i++) {
-            timeline.addEvent(mkMessage());
+            timeline.addEvent(mkMessage({msg: ""+i}));
```

Comment:
I assume you're using numbers in the `content.body` to aid debugging. It may be clearer to just put a tiny prefix here to make it clear that this is the textual body and not say, the index of the event.